backend/station_manager/app/src/test/java/com/StationManager/app/domain/trainstation/HallTest.java

```diff
@@ -0,0 +1,698 @@
+                new Hall(
+                        new Position(new Point(0, 0), new Point(20, 20)), entrances, ticketOffices);
+
+        hall.addTicketOffice(ticketOffice1);
+        hall.addTicketOffice(ticketOffice2);
+        hall.addTicketOffice(ticketOffice3);
+
+        assertEquals(3, hall.getTicketOffices().size());
+    }
+
+    @Test
+    @DisplayName("Adding TicketOffice when Position is taken. Should throw exception")
+    void testAddTicketOfficeToHallFailsWhenAddingOnTakenPosition() {
+        ArrayList<Position> entrances = new ArrayList<>();
+        ArrayList<TicketOffice> ticketOffices = new ArrayList<>();
+
+        Hall hall =
+                new Hall(
+                        new Position(new Point(0, 0), new Point(20, 20)), entrances, ticketOffices);
+
+        Position ticketOfficePosition1 = new Position(new Point(0, 0), new Point(2, 1));
+        Position ticketOfficePosition2 = new Position(new Point(0, 0), new Point(2, 1));
+
+        TicketOffice ticketOffice1 = new TicketOffice(ticketOfficePosition1, Direction.Down);
+        TicketOffice ticketOffice2 = new TicketOffice(ticketOfficePosition2, Direction.Down);
+
+        hall.addTicketOffice(ticketOffice1);
+
+        assertThrows(IllegalStateException.class, () -> hall.addTicketOffice(ticketOffice2));
+    }
+
+    @Test
+    @DisplayName("Adding client to single ticket office in up")
+    void testAddClientToSingleUpTicketOffice() {
+        ArrayList<Position> entrances = new ArrayList<>();
+        ArrayList<TicketOffice> ticketOffices = new ArrayList<>();
+
+        Hall hall =
+                new Hall(
+                        new Position(new Point(0, 0), new Point(20, 20)), entrances, ticketOffices);
+
+        Position ticketOfficePosition1 = new Position(new Point(10, 9), new Point(12, 10));
+        TicketOffice ticketOffice1 = new TicketOffice(ticketOfficePosition1, Direction.Up);
+
+        hall.addTicketOffice(ticketOffice1);
+        ticketOffice1.setClosed(false);
+
+        Client client1 =
+                new Client(
+                        2,
+                        "Fname",
+                        "Sname",
+                        (short) 2,
+                        new Privilegy("ordinary", 0),
+                        new Point(3, 4));
+
+        hall.addClient(client1);
+
+        assertEquals(
+                ticketOffice1.getQueue().getClients().get(0),
+                new Client(
+                        2,
+                        "Fname",
+                        "Sname",
+                        (short) 2,
+                        new Privilegy("ordinary", 0),
+                        new Point(11, 11)));
+    }
+
+    @Test
+    @DisplayName("Adding client to single ticket office in down")
+    void testAddClientToSingleDownTicketOffice() {
+        ArrayList<Position> entrances = new ArrayList<>();
+        ArrayList<TicketOffice> ticketOffices = new ArrayList<>();
+
+        Hall hall =
+                new Hall(
```

Comment:
format this manually to be more readable. will discuss formatters later on

Reply:
Here we need to completely disable the formatter before the commit? Because this is exactly how he formats this code.

Reply:
yeap